src/server.cpp

```diff
@@ -104,25 +104,44 @@ void *ServerThread::run()
 	/*
 	 * The real business of the server happens on the ServerThread.
 	 * How this works:
-	 * AsyncRunStep() runs an actual server step as soon as enough time has
-	 * passed (dedicated_server_loop keeps track of that).
-	 * Receive() blocks at least(!) 30ms waiting for a packet (so this loop
-	 * doesn't busy wait) and will process any remaining packets.
+	 * AsyncRunStep() (which runs the actual server step) is called at the
+	 * server-step frequency. Receive() is used for waiting between the steps.
 	 */
 
+	const float dedicated_server_step = g_settings->getFloat("dedicated_server_step");
+	const float singleplayer_server_step = 1.0f / g_settings->getFloat("fps_max");
```

Comment:
Needs limiting. See `FpsControl::limit` as an example. Same for the unfocused FPS.